bootstrap/kubeadm/internal/cloudinit/kubeadm-bootstrap-script.sh

```diff
@@ -127,11 +127,25 @@ retry-command kubeadm join phase control-plane-prepare certs
 retry-command kubeadm join phase control-plane-prepare kubeconfig
 retry-command kubeadm join phase control-plane-prepare control-plane
 # {{ end }}
+
+# Check if the ControlPlaneKubeletLocalMode feature-gate is enabled
+CONTROLPLANE_KUBELET_LOCAL_MODE="false"
+kubectl get configmap -n kube-system kubeadm-config -o jsonpath='{.data.ClusterConfiguration}' | \
+  grep -q 'ControlPlaneKubeletLocalMode: true' > /dev/null && CONTROLPLANE_KUBELET_LOCAL_MODE="true"
+
 retry-command kubeadm join phase kubelet-start
-# {{ if .ControlPlane }}
-try-or-die-command kubeadm join phase control-plane-join etcd
-retry-command kubeadm join phase control-plane-join update-status
-retry-command kubeadm join phase control-plane-join mark-control-plane
-# {{ end }}
+
+if [[ "${CONTROLPLANE_KUBELET_LOCAL_MODE}" == "true" ]]; then
+  # Run kubeadm join and skip all already executed phases
+  # kubeadm join relies on hidden phases when CONTROLPLANE_KUBELET_LOCAL_MODE. These cannot be run
+  # externally.
```

Comment:
We can even think about not doing this if/else.

We would only loose that we don't retry the `kubeadm join phase control-plane-join mark-control-plane`.